syntax = "proto3";

package vio_input_proto;

message Vec3 {
  double x = 1;
  double y = 2;
  double z = 3;
}

message IMUData
{
  fixed64 timestamp = 1;
  Vec3 angular_vel = 2;
  Vec3 linear_accel = 3;
}

message CamData {
  fixed64 timestamp = 1;
  int32 rows = 2;
  int32 cols = 3;
  bytes img0_data = 4;
  bytes img1_data = 5;
}

message IMUCamVec {
  repeated IMUData imu_data = 1;
  CamData cam_data = 2;
  fixed64 real_timestamp = 3;
  int32 frame_id = 4;
}